pkg/reconciler/awssnssource/reconciler_test.go

```diff
@@ -113,3 +133,326 @@ func newEventSource() *v1alpha1.AWSSNSSource {
 
 	return src
 }
+
+// TestReconcileSubscription contains tests specific to the SNS source.
+func TestReconcileSubscription(t *testing.T) {
+	testCases := rt.TableTest{
+		// Regular lifecycle
+
+		{
+			Name:          "Not yet subscribed",
+			Key:           tKey,
+			OtherTestData: makeMockSubscriptionsPages(false),
+			Objects: []runtime.Object{
+				newReconciledSource(),
+				newReconciledAdapter(),
+			},
+			WantStatusUpdates: []clientgotesting.UpdateActionImpl{{
+				Object: newReconciledSource(subscribed),
+			}},
+			WantEvents: []string{
+				subscribedEvent(),
+			},
+			PostConditions: []func(*testing.T, *rt.TableRow){
+				calledSubscribe(true),
+			},
+		},
```

Comment:
The actual test suite is here. Those are usual Knative table tests, the only difference with the "common" test suite is that all tests here assume the receive adapter was already reconciled, so we can focus purely on the reconciliation of the SNS subscription.